src/_guides/libraries/create-library-packages.md

```diff
@@ -153,6 +154,63 @@ it could instead use the `package:` directive
 </aside>
 
 
+## Conditionally importing and exporting library files
+
+If your library supports multiple platforms,
+then you might need to conditionally import or export library files.
+A common use case is a library that supports both web and native platforms.
+
+To conditionally import or export,
+you must check for the presence of `dart:*` libraries.
+Here's an example of conditional export code that
+checks for the presence of `dart:io` and `dart:html`:
```

Comment:
nit: I would perhaps consider testing on `dart:js` instead of `dart:html` because I would imagine `dart:js` is more comon in Flutter for web packages (just a guess).

Reply:
I've seen dart:html in the wild, but that's probably pre-flutter web, and I'd love to get a definitive answer on this. @johnpryan?

Reply:
It depends on what platform library you actually depend on.
This is *library* testing, not *platform* testing&mdash;which is why the example is actually not very good. It's not detecting "VM" or "JavaScript", it's detecting the presence of the `dart:io` library or the `dart:html` library.

I know of at least one experiment with embedding the VM and a browser engine in the same program, and that platform hae *both* `dart:io` and `dart:html` (not sure if it had `dart:js` though, since you used Dart instead).

So, if you depend on `dart:js` then test for `dart.library.js`, and if you depend on `dart:html` then test for `dart.library.html`. That makes your code maximally supported. 

In most cases, the feature you need is a DOM feature, not a plain JS feature, so I'd test for `dart.library.html`.

Reply:
What happens if both dart:io & dart:html are supported? Would only the first `if` take effect, or might you import multiple libraries? (I'd hope it's the former, not the latter.)